tests/credential_plugins_test.py

```diff
@@ -134,6 +135,67 @@ def test_hashivault_handle_auth_not_enough_args() -> None:
         hashivault.handle_auth()
 
 
+@pytest.mark.parametrize(
+    'kwargs',
+    (
+        {
+            'access_key': 'my_access_key',
+            'secret_key': 'my_secret_key',
+            'role_arn': 'the_arn',
+            'identifier': 'access_token',
+        },
+    ),
+    (
+        {
+            'role_arn': 'the_arn',
+            'identifier': 'access_token',
+        },
+    ),
+)
+@pytest.mark.parametrize(
+    (
+        'identifier_key',
+        'expected',
+    ),
+    (
+        (None, 'the_access_token'),
+        ('access_key', 'the_access_key'),
+        ('secret_key', 'the_secret_key'),
+    ),
+)
+def test_aws_assumerole_identifier(
+    monkeypatch: pytest.MonkeyPatch,
+    kwargs: dict, identifier_key: str, expected: dict[str, str],
+) -> None:
+    """Test that the aws_assumerole_backend function call returns a token given
+    the access_key and secret_key."""
+
+    def mock_getcreds(
+            access_key: str | None,
+            secret_key: str | None,
+            role_arn: str | None,
+            external_id: int,
+    ) -> dict:
+        return {
+            'access_key': 'the_access_key',
+            'secret_key': 'the_secret_key',
+            'access_token': 'the_access_token',
+            'Expiration': datetime.datetime.today() + datetime.timedelta(days=1),
+        }
+
+    monkeypatch.setattr(
+        aws_assumerole,
+        'aws_assumerole_getcreds',
+        mock_getcreds,
+    )
+
+    if identifier_key:
+        kwargs['identifier'] = identifier_key
+
+    token = aws_assumerole.aws_assumerole_backend(**kwargs)
```

Comment:
I think it's cleaner to have an additional var here instead of mutating the inputs:
```suggestion
    extra_kwargs = {'identifier': identifier_key} if identifier_key else {}

    token = aws_assumerole.aws_assumerole_backend(**kwargs, **extra_kwargs)
```